app/src/org/commcare/views/widgets/DocumentWidget.java

```diff
@@ -0,0 +1,72 @@
+package org.commcare.views.widgets;
+
+import android.content.ActivityNotFoundException;
+import android.content.Context;
+import android.view.View;
+import android.widget.Toast;
+
+import com.google.android.material.button.MaterialButton;
+
+import org.commcare.activities.components.FormEntryConstants;
+import org.commcare.dalvik.R;
+import org.commcare.logic.PendingCalloutInterface;
+import org.commcare.utils.StringUtils;
+import org.javarosa.form.api.FormEntryPrompt;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+public class DocumentWidget extends MediaWidget {
+
+    public DocumentWidget(Context context, final FormEntryPrompt prompt, PendingCalloutInterface pic) {
+        super(context, prompt, pic);
+    }
+
+    @Override
+    protected void initializeButtons() {
+        mChooseButton = new MaterialButton(getContext());
+        WidgetUtils.setupButton(mChooseButton,
+                StringUtils.getStringSpannableRobust(getContext(), R.string.choose_document),
+                !mPrompt.isReadOnly());
+
+        mChooseButton.setOnClickListener(v -> {
+            try {
+                ((AppCompatActivity)getContext())
+                        .startActivityForResult(WidgetUtils.createPickMediaIntent(getContext(), "application/*,text/*"),
+                                FormEntryConstants.AUDIO_VIDEO_FETCH);
+                pendingCalloutInterface.setPendingCalloutFormIndex(mPrompt.getIndex());
+            } catch (ActivityNotFoundException e) {
+                Toast.makeText(getContext(),
+                        StringUtils.getStringSpannableRobust(getContext(),
+                                R.string.activity_not_found, "choose document"),
+                        Toast.LENGTH_SHORT).show();
+            }
+        });
+
+        String acq = mPrompt.getAppearanceHint();
+        if (QuestionWidget.ACQUIREFIELD.equalsIgnoreCase(acq)) {
+            mChooseButton.setVisibility(View.GONE);
+        }
+    }
+
+    @Override
+    protected void setupLayout() {
+        addView(mChooseButton);
+    }
+    
+    @Override
+    public void setOnLongClickListener(OnLongClickListener l) {
+        mChooseButton.setOnLongClickListener(l);
+    }
+
+    @Override
+    public void unsetListeners() {
+        superUnsetListeners();
+        mChooseButton.setOnLongClickListener(null);
+    }
+
+    @Override
+    public void cancelLongPress() {
+        superCancelLongPress();
+        mChooseButton.cancelLongPress();
+    }
```

Comment:
@minhaminha think we should just do `super. unsetListeners` directly here and have the super class (MediaWidget) implement null checks on individual buttons to supprt these calls.

Reply:
Is this not simpler? Otherwise, I'd have to do null checks on at least the `mCapture` and `mPlay` buttons for 3 different functions, which I think would clutter `MediaWidget`, when all it really needs to do is call the grandfather method in QuestionWidget. Would renaming the function to make this clearer help?

Reply:
Hmm I think it's pretty bad practie in Java to bypass super method chain like this (quite unusual I would say) and if we are having to do that, I don't think `MediaWidget` in it's current form is the right choice for super class for `documentWidget`. `MediaWidget` is designed to support media playback with the ability to play, capture and add media from file provider. It seems like we only need one of these functionalities (add file from file provider here) , So think we need to either modify `MediaWidget` to be more flexible by allowing for these use cases by expecting buttons as optional OR have a different `FileChooserWidget` that abstracts the functionality of "adding a file" to a common place and can be further utilized as a super class by both `MediaWidget` and `DocumentWidget`.

Reply:
That's fair! I went with the first option after realizing that it would actually be less lines of code overall. I was also a bit confused about which `setOnLongClickListener` QuestionWidget would call (I thought it would call MediaWidget) but it does in fact, use the DocumentWidget's, which is good to know.